arangod/Cluster/ClusterMethods.cpp

```diff
@@ -3301,24 +3301,49 @@ arangodb::Result matchBackupServersSlice(VPackSlice const planServers,
   match.clear();
 
   // Local copy of my servers
-  std::unordered_set<std::string> localCopy;
+  // Note that we use a `std::set` here for the following reasons:
+  //  - this function is supposed to return a canonical result, which
+  //    does not depend on the order of the input in `planServers` or
+  //    `dbServers`, therefore we use a sorted set here and a sorted
+  //    map as the result.
+  //  - Performance does not matter at all, this method is called for
+  //    hotbackup download and hotbackup restore, both methods are
+  //    using considerable amounts of time. The server lists are usually
+  //    quite small (3 or 5 or a few dozen at most).
+  // So please do not "optimize" this.
+  std::set<std::string> localCopy;
```

Comment:
std::less<> as second parameter, in such case possible to find string_view without unnecessary conversation: is_transparent
https://en.cppreference.com/w/cpp/utility/functional/less_void

Reply:
Sorry, I do not understand this. According to cppreference the second template argument of `std::set` is by default `std::less`. Please elaborate.

Reply:
> According to cppreference the second template argument of std::set is by default std::less. Please elaborate.

Not really :(

https://stackoverflow.com/questions/67456235/why-does-the-c-standard-not-change-stdset-to-use-stdless-as-its-default

Reply:
OK, understood. Thanks for the hint, but I will ignore it. As argued previously, this is not a performance relevant place at all. Lists of servers are typically size 3, in very rare cases they are 20. And we are in the context of a backup download or restore, which are both relatively slow operations. So it is not worthwhile to
 - clutter the code with useless optimizations
 - make the reader think about why this particular optimization was done here
Finally, the linked article even contains this:
"Moving from std::set<T, std::less<T>> to std::set<T, std::less<>> can make key-finding algorithms more efficient, if the search-key never has to be converted.

Conversely, it can make them less efficient if the conversion occurrs on every call to the comparator, instead of once in the caller on starting the algorithm. Those conversions can be quite expensive."
So without measurement we should not embark on this. And since sizes are small, I doubt it that we can even perform a sensible measurement.

Reply:
> Lists of servers are typically size 3, in very rare cases they are 20.

It's not about size of elements in map, it's about unnecessary construction of string (copy and allocation) from string_view (commonly can be received from vpack/constant) in case of find in this map.

> "Moving from std::set<T, std::less> to std::set<T, std::less<>> can make key-finding algorithms more efficient, if the search-key never has to be converted.

For find string_view in map/set of strings it's never converting.

Single conversation (same as with default less but not required in case of inserting existing key) will be in case of insert/emplace


Anyway, my comment not required you to do something, but it's useful tip, in c++ for map/set strings almost always better to write less<>.

It probably could be changed in standard if not ABI issues, but because of ABI we should make it explicit default.